src/interpreter_funcs.rs

```diff
@@ -77,11 +78,50 @@ impl Func for FuncImplShrinkWrap {
     }
 }
 
+pub struct FuncImplLaplacianSmoothing;
+impl Func for FuncImplLaplacianSmoothing {
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
```

Comment:
Just a note. I actually spent a non-trivial amount of time on this review making sure this holds 😂 These things are tricky.

Reply:
so, is it correct or not?